plugins/modules/agent_job_category.ps1

```diff
@@ -92,8 +92,7 @@ try {
             $module.Result.changed = $true
         }
     }
-    $outputHash = ConvertTo-HashTable -Object $output
-    $module.Result.data = $outputHash
+    $module.Result.data = Format-JsonOutput -Object $output
```

Comment:
Does this return correctly? Usually the result data is meant to be an object because `$module` will serialize it, so I would think that doing it this way is going to serialize the JSON string itself.

An option may be to make the function just format objects (correct properties, etc.) that will serialize properly. There's an example I did in `win_psmodulei_info` that does that because there are so many problematic fields.

Reply:
That was a great example to start off of. I can't even recall now what made me go with the original Hash Table route honestly, but this feels a lot cleaner! 🙇🏻